flaky  test_that_multiple_everest_clients_can_connect_to_server

https://github.com/equinor/ert/actions/runs/13805674539/job/38615900947
https://github.com/equinor/ert/actions/runs/13829457140/job/38690536470

Two distinct failures:
1 (https://github.com/equinor/ert/actions/runs/13805674539/job/38615900947):
```
FAILED tests/everest/test_everest_client.py::test_that_multiple_everest_clients_can_connect_to_server - assert False
 +  where False = all(<generator object test_that_multiple_everest_clients_can_connect_to_server.<locals>.<genexpr> at 0x12e104930>)
```
for mismatching runmodel event lists between clients listening to the same server

2 (https://github.com/equinor/ert/actions/runs/13829457140/job/38690536470):
```
FAILED tests/everest/test_everest_client.py::test_that_multiple_everest_clients_can_connect_to_server - AssertionError: assert ['EverestStat...ltEvent', ...] == ['EverestStat...ltEvent', ...]
  
  Right contains one more item: 'EverestBatchResultEvent'
  
  Full diff:
    [
        'EverestStatusEvent',
        'EverestBatchResultEvent',
        'EverestBatchResultEvent',
        'EverestBatchResultEvent',
        'EverestStatusEvent',
        'EverestBatchResultEvent',
        'EverestBatchResultEvent',
        'EverestStatusEvent',
        'EverestBatchResultEvent',
        'EverestBatchResultEvent',
        'EverestStatusEvent',
        'EverestBatchResultEvent',
  -     'EverestBatchResultEvent',
    ]
```
different everest events recorded on server vs on client

--- a/src/ert/run_models/everest_run_model.py
+++ b/src/ert/run_models/everest_run_model.py
@@ -248,7 +248,11 @@ class EverestRunModel(BaseRunModel):
             failed = True
             exception = e
         finally:
-            if self._exit_code != EverestExitCode.COMPLETED:
+            if self._exit_code not in {
+                EverestExitCode.COMPLETED,
+                EverestExitCode.MAX_FUNCTIONS_REACHED,
+                EverestExitCode.MAX_BATCH_NUM_REACHED,
+            }:
                 failed = True
             self.send_event(
                 EndEvent(

--- a/src/everest/detached/jobs/everserver.py
+++ b/src/everest/detached/jobs/everserver.py
@@ -152,6 +152,8 @@ class ExperimentRunner:
                     sub.notify()
 
                 if isinstance(item, EndEvent):
+                    # Wait for subscribers to receive final events
+                    await asyncio.sleep(5)
                     break
                 await asyncio.sleep(0.1)
             await simulation_future

--- a/tests/everest/test_everest_client.py
+++ b/tests/everest/test_everest_client.py
@@ -11,7 +11,7 @@ from starlette.responses import Response
 
 from everest.bin.everest_script import everest_entry
 from everest.config import EverestConfig, ServerConfig
-from everest.detached import wait_for_server
+from everest.detached import server_is_running
 from everest.detached.jobs.everserver import _find_open_port
 from everest.gui.everest_client import EverestClient
 from everest.strings import STOP_ENDPOINT
@@ -160,7 +160,13 @@ def test_that_multiple_everest_clients_can_connect_to_server(cached_example):
     )
 
     everest_main_thread.start()
-    wait_for_server(ever_config.output_dir, 60)
+
+    def everserver_is_running():
+        return server_is_running(
+            *ServerConfig.get_server_context(ever_config.output_dir)
+        )
+
+    wait_until(everserver_is_running, interval=1, timeout=20)
 
     server_context = ServerConfig.get_server_context(ever_config.output_dir)
     url, cert, auth = server_context

